src/posix.zig

```diff
@@ -21,17 +22,181 @@ const cstd = struct {
 
 const trace = @import("trace.zig");
 
-const global = struct {
-    export var optarg: [*:0]u8 = undefined;
-    export var opterr: c_int = undefined;
-    export var optind: c_int = 1;
-    export var optopt: c_int = undefined;
+pub const GetoptContext = extern struct {
+    optarg: ?[*:0]const u8 = null,
+    opterr: c_int = 1,
+    optind: c_int = 1,
+    optopt: c_int = undefined,
+    // When an element of `argv` contains multiple option characters, it is unspecified how `getopt`
+    // determines which options have already been processed.
+    //
+    // so we use an additional cursor variable
+    optcur: c_int = 0,
 };
 
+export fn z_getopt_r(context: *GetoptContext, argc: c_int, argv: [*]const ?[*:0]const u8, optstring: [*:0]const u8) callconv(.C) c_int {
```

Comment:
why is this being exported?

Reply:
to make a reentrant version available to the user.

Reply:
Is `z_getopt_r` a part of libc?

Reply:
yes, it should be exposed to the user as a normal part of libc.

Reply:
> yes, it should be exposed to the user as a normal part of libc

What I'm asking is, which standard libc is this apart of?  The cstd, glibc, musl, msvc.  The purpose of ziglibc is to provide an alternative implementation for existing libc libraries.